src/cockpit/channels/filesystem.py

```diff
@@ -200,12 +247,14 @@ async def set_contents(
                 # no preconditions about what currently exists or not
                 # calculate the file mode from the umask
                 os.fchmod(fd, 0o666 & ~my_umask())
+                chown_if_required(fd)
```

Comment:
This only applies the chown logic for "nonexisting file". But it also needs to be applied to the "keep existing owner/group" logic below at line 268 from commit 72027c1d3d5c, in the "else:" branch.

Reply:
How did I overlook this :)

So I think this would be the last behavior to be made, I somehow completely missed this..